### ISSUE TITLE
[Feature]: Merge download buttons into 1

### ISSUE THREAD

**Issue by ikanakova** [fast4x/RiMusic#464]
### Description

Merge the download buttons into one.
<img src="https://github.com/fast4x/RiMusic/assets/38765251/0dd8b476-89bf-49ae-8296-2eaf6e3be596" width="500">

### Suggested Solution

Merge the download buttons into one so that when you press everything it will download or if it's already downloaded it will remove everything. (If someone wants to download or delete individual songs, they should do it for that specific song.) When 1 song is downloaded in the playlist, the playlist appears to be downloaded. And when the user clicks on the download button for the whole playlist, it will ask if the user really wants to delete all the downloaded ones, and if he answers yes, the whole playlist will be deleted; if there is no song in the playlist, it will ask him if he really wants to download the whole playlist...

### Additional Context

Related to #149 https://github.com/fast4x/RiMusic/issues/368#issuecomment-1882738030

**Comment by sayyid5416** [comment]
yes, it would be good for app refinement. Simplicity of features is always better. Less buttons = less complexity. I like this idea.

**Comment by ikanakova** [comment]
Now that there is a new icon for downloaded, I think it's even more confusing to have two icons (downloaded and not downloaded) next to each other. At least it looks that way to me from the screenshots in the issue  #1249 (I don't have the new version yet, because I have the F-Droid version).
